### PR TITLE
Beacd 58

### PR DESCRIPTION
**Pull request by gaurav-hmrc**

### CODE REVIEW

File: app/models/employer/DoesBusinessManagePAYE.scala [hunk 1]
```diff
@@ -14,10 +14,13 @@
  * limitations under the License.
  */
 
-package models.employer.ers
+package models.employer
 
 import utils.{Enumerable, RadioOption, WithName}
 
+/**
```
Comment (prashantchoudhari) [inline app/models/employer/DoesBusinessManagePAYE.scala]: we need to remove this

File: app/models/employer/IsBusinessRegisteredForPAYE.scala [hunk 2]
```diff
@@ -14,10 +14,13 @@
  * limitations under the License.
  */
 
-package models.employer.cis.uk.contractor
+package models.employer
 
 import utils.{Enumerable, RadioOption, WithName}
 
+/**
```
Comment (prashantchoudhari) [inline app/models/employer/IsBusinessRegisteredForPAYE.scala]: we need to remove this

File: app/controllers/employer/cis/uk/contractor/DoesBusinessManagePAYEController.scala [hunk 3]
```diff
@@ -47,6 +47,8 @@ class DoesBusinessManagePAYEController @Inject()(
   lazy val viewAction = ViewAction(routes.DoesBusinessManagePAYEController.onSubmit(), "CisUkContractorEpaye")
   val form = formProvider()
 
+  lazy val action = ViewAction(routes.DoesBusinessManagePAYEController.onSubmit(), "CisUkContractorEpaye")
```
Comment (kurtlogan) [inline app/controllers/employer/cis/uk/contractor/DoesBusinessManagePAYEController.scala]: Is this line used? The viewAction above seems to be used

File: app/controllers/employer/cis/uk/contractor/IsBusinessRegisteredForPAYEController.scala [hunk 4]
```diff
@@ -47,6 +48,8 @@ class IsBusinessRegisteredForPAYEController @Inject()(
   lazy val viewAction = ViewAction(routes.IsBusinessRegisteredForPAYEController.onSubmit(), "AddCisUkContractor")
   val form = formProvider()
 
+  lazy val action = ViewAction(routes.IsBusinessRegisteredForPAYEController.onSubmit(), "CisUkContractorEpaye")
```
Comment (kurtlogan) [inline app/controllers/employer/cis/uk/contractor/IsBusinessRegisteredForPAYEController.scala]: Is this used? The viewAction above seems to be used

File: app/controllers/employer/intermediaries/DoesBusinessManagePAYEController.scala [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+  val form = formProvider()
+
+  lazy val action = ViewAction(routes.DoesBusinessManagePAYEController.onSubmit(), "AddIntermediariesEpayeOnline")
+
+  def onPageLoad() = (authenticate andThen serviceInfoData) { implicit request =>
+    Ok(doesBusinessManagePAYE(appConfig, form, action)(request.serviceInfoContent))
+  }
+
+  def onSubmit() = (authenticate andThen serviceInfoData).async { implicit request =>
```
Comment (kurtlogan) [inline app/controllers/employer/intermediaries/DoesBusinessManagePAYEController.scala]: Is async needed in this context?

File: app/controllers/employer/intermediaries/IsBusinessRegisteredForPAYEController.scala [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+  val form = formProvider()
+
+  lazy val action = ViewAction(routes.IsBusinessRegisteredForPAYEController.onSubmit(), "AddIntermediaries")
+
+  def onPageLoad() = (authenticate andThen serviceInfoData) { implicit request =>
+    Ok(isBusinessRegisteredForPAYE(appConfig, form, action)(request.serviceInfoContent))
+  }
+
+  def onSubmit() = (authenticate andThen serviceInfoData).async { implicit request =>
```
Comment (kurtlogan) [inline app/controllers/employer/intermediaries/IsBusinessRegisteredForPAYEController.scala]: Is async needed here?

File: app/utils/nextpage/employer/IsBusinessRegisteredForPAYENextPage.scala [hunk 7]
```diff
@@ -39,4 +41,32 @@ trait IsBusinessRegisteredForPAYENextPage {
     }
   }
 
+  implicit val isBusinessRegisteredForPAYEEI
```
Comment (kurtlogan) [inline app/utils/nextpage/employer/IsBusinessRegisteredForPAYENextPage.scala]: naming is different from below val, naming conventions should be the same. Naming should be `eiIsBusinessRegisteredForPAYE` as per other files and methods

File: app/views/employer/intermediaries/usePAYEEmployerAccount.scala.html [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+           class="button"
+           data-journey-click="AddIntermediariesOtherAccount:Click:SignIn">
+
+            @messages("usePAYEEmployerAccount.continue")
+        </a>
+    </p>
+
+    <p>
+        <a id="sign-in"
+           href="/employment-intermediary-report/not-enrolled"
```
Comment (kurtlogan) [inline app/views/employer/intermediaries/usePAYEEmployerAccount.scala.html]: We shouldn't be using fixed urls, this should be in the config

File: conf/messages.cy [hunk 9]
```diff
@@ -611,3 +611,18 @@ registerEmployersPAYE.subtext1 = Mae angen i chi fod wedi cofrestru ar gyfer TWE
...
 registerEmployersPAYE.notnow = Nid wyf eisiau gwneud hyn ar hyn o bryd
+
+#######################################################
+##  UsePAYEEmployerIntermediariesAccount
+#######################################################
+usePAYEEmployerIntermediariesAccount.addAccount = Rwyf eisiau ychwanegu cyfryngwyr yn y cyfrif hwn
+usePAYEEmployerIntermediariesAccount.p1 = Dylech reoli cyfryngwyr cyflogaeth yn yr un cyfrif ag yr ydych yn rheoli TWE i gyflogwyr.
+usePAYEEmployerIntermediariesAccount.p2 = Bydd hynny’n ei wneud yn haws rheoli’r ddau gynllun.
+
+
```
Comment (kurtlogan) [inline conf/messages.cy]: Remove extra white space

File: conf/messages.en [hunk 10]
```diff
@@ -638,3 +638,17 @@ registerEmployersPAYE.subtext1 = You need to be registered for PAYE for employer
...
 registerEmployersPAYE.notnow = I don’t want to do this right now
+
+#######################################################
+##  UsePAYEEmployerAccountIntermediaries
+#######################################################
+usePAYEEmployerIntermediariesAccount.addAccount = I want to add intermediaries in this account
+usePAYEEmployerIntermediariesAccount.p1 = You should manage employment intermediaries in the same account that you manage PAYE for employers.
+usePAYEEmployerIntermediariesAccount.p2 = It will make it easier to manage both schemes.
+
+
```
Comment (kurtlogan) [inline conf/messages.en]: Remove extra whitespace

**Comment by prashantchoudhari** [comment]
@gaurav-hmrc  - it needs merging master.. rest I think was fine!

**Comment by gaurav-hmrc** [comment]
review comments implemented

File: app/views/employer/intermediaries/usePAYEEmployerAccount.scala.html [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+@main_template(
+    title = messages("usePAYEEmployerAccount.title"),
+    appConfig = appConfig,
+    serviceInfoContent = serviceInfoContent
+) {
+
+    @components.heading("usePAYEEmployerAccount.heading")
+
+    <p>@messages("usePAYEEmployerIntermediariesAccount.p1")</p>
```
Comment (kurtlogan) [inline app/views/employer/intermediaries/usePAYEEmployerAccount.scala.html]: Why does this use a different key?

**Pull request merged**